File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/networks/model/Network.java
package dev.reuss.tmdb.domain.networks.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * TMDB network details.
 *
 * @param headquarters  network headquarters
 * @param homepage      network homepage URL
 * @param id            TMDB network id
 * @param logoPath      TMDB logo image path
 * @param name          network name
 * @param originCountry origin country code
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record Network(
        String headquarters,

        String homepage,

        int id,

        @JsonProperty("logo_path")
        String logoPath,

        String name,

        @JsonProperty("origin_country")
        String originCountry
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/networks/model/Network.kt
package dev.reuss.tmdb.domain.networks.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * TMDB network details.
 *
 * @property headquarters network headquarters
 * @property homepage network homepage URL
 * @property id TMDB network id
 * @property logoPath TMDB image path for the logo.
 * @property name network name
 * @property originCountry ISO 3166-1 code for the resource's country of origin.
 */
@JvmRecord
data class Network(
    @all:JsonProperty("headquarters")
    val headquarters: String?,
    @all:JsonProperty("homepage")
    val homepage: String?,
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("logo_path")
    val logoPath: String?,
    @all:JsonProperty("name")
    val name: String?,
    @all:JsonProperty("origin_country")
    val originCountry: String?,
) : TmdbModel
